src/pages/studyView/resources/FilesAndLinks.tsx

```diff
@@ -0,0 +1,329 @@
+} from 'shared/components/lazyMobXTable/LazyMobXTable';
+import { observer } from 'mobx-react';
+import _, { forEach } from 'lodash';
+import internalClient from 'shared/api/cbioportalInternalClientInstance';
+import { Else, If, Then } from 'react-if';
+import { WindowWidthBox } from '../../../shared/components/WindowWidthBox/WindowWidthBox';
+import LoadingIndicator from 'shared/components/loadingIndicator/LoadingIndicator';
+import {
+    getSampleViewUrlWithPathname,
+    getPatientViewUrlWithPathname,
+} from 'shared/api/urls';
+import { getAllClinicalDataByStudyViewFilter } from '../StudyViewUtils';
+import { StudyViewPageStore } from 'pages/studyView/StudyViewPageStore';
+import { isUrl, remoteData } from 'cbioportal-frontend-commons';
+import { makeObservable, observable, computed } from 'mobx';
+import { ResourceData, StudyViewFilter } from 'cbioportal-ts-api-client';
+
+export interface IFilesLinksTable {
+    store: StudyViewPageStore;
+}
+
+class FilesLinksTableComponent extends LazyMobXTable<{
+    [id: string]: string;
+}> {}
+
+const RECORD_LIMIT = 500;
+
+async function fetchResourceDataOfPatient(patientIds: Map<string, string>) {
+    const ret: { [key: string]: ResourceData[] } = {};
+    const promises = [];
+    for (let [patientId, studyId] of patientIds) {
+        promises.push(
+            internalClient
+                .getAllResourceDataOfPatientInStudyUsingGET({
+                    studyId: studyId,
+                    patientId: patientId,
+                    projection: 'DETAILED',
+                })
+                .then(data => {
+                    if (patientId in ret) {
+                        ret[patientId].push(...data);
+                    } else {
+                        ret[patientId] = data;
+                    }
+                })
+        );
+    }
+
+    return Promise.all(promises).then(() => ret);
+}
+
+async function fetchFilesLinksData(
+    filters: StudyViewFilter,
+    sampleIdResourceData: { [sampleId: string]: ResourceData[] },
+    searchTerm: string | undefined,
+    sortAttributeId: string | undefined,
+    sortDirection: 'asc' | 'desc' | undefined,
+    recordLimit: number
+) {
+    let sampleClinicalDataResponse = await getAllClinicalDataByStudyViewFilter(
+        filters,
+        searchTerm,
+        sortAttributeId,
+        sortDirection,
+        recordLimit,
+        0
+    );
+
+    const patientIds = new Map<string, string>();
+
+    forEach(sampleClinicalDataResponse.data, (data, uniqueSampleId) => {
+        forEach(data, item => {
+            patientIds.set(item.patientId, item.studyId);
+        });
+    });
+
+    const resourcesPerPatient: { [key: string]: number } = {};
+    const items: { [attributeId: string]: any } = [];
+
+    const itemBuilder = (resourceData: { [key: string]: ResourceData[] }) => {
+        forEach(resourceData, (data: ResourceData[]) => {
+            forEach(data, (resource: ResourceData) => {
+                items.push({
```

Comment:
It's not ideal to modify the variables outside the function. it would be better if this function returned `items`.

Reply:
The reason of this implementation is performance and avoiding merging multiple lists. Otherwise, we need to merge multiple arrays before returning the results, that is - the merge will cost O(n).

Reply:
@Nelliney at the relatively small scale of our data, the cost of iteration is very low, almost negligible.  so we generally err on the side of readability unless we can demonstrate there is a performance issue.  for example something like this would be more easy to understand and functional:
```
const things = _(resourceData).values().flatMap(resource=>{ .... }).value();
```
now, in one line, I can understand what we are doing.    i don't have to wrap my head around "itemBuilder"

if you need to iterate through the list AGAIN to accomplish what's at line 98, that's fine.

Reply:
Done